helix-core/src/test/java/org/apache/helix/common/ZkTestBase.java

```diff
@@ -154,8 +215,29 @@ public void afterSuite() throws IOException {
       }
     }
 
+    // Close ZK-related resources
+    _gSetupTool.close();
+    _baseAccessor.close();
     _gZkClient.close();
     TestHelper.stopZkServer(_zkServer);
+
+    // If there are multiple ZooKeepers, close them all
+    if (System.getProperty(MULTI_ZK_PROPERTY_KEY) != null) {
+      for (int i = 0; i < _numZk; i++) {
+        if (_baseDataAccessorMap != null && _baseDataAccessorMap.containsKey(i)) {
+          _baseDataAccessorMap.get(i).close();
+        }
+        if (_clusterSetupMap != null && _clusterSetupMap.containsKey(i)) {
+          _clusterSetupMap.get(i).close();
+        }
+        if (_helixZkClientMap != null && _helixZkClientMap.containsKey(i)) {
+          _helixZkClientMap.get(i).close();
+        }
+        if (_zkServerMap != null && _zkServerMap.containsKey(i)) {
+          TestHelper.stopZkServer(_zkServerMap.get(i));
+        }
```

Comment:
Nit: could use `values()` to close without `get()` and checking every time.

Reply:
Null checks are necessary, and the rest is a matter of style.